src/r_main.c

```diff
@@ -687,12 +687,31 @@ void R_ExecuteSetViewSize (void)
 
 void R_Init (void)
 {
+  // [Nugget] Dynamic arrays
+  static boolean first_allocation = true;
+  
+  #define width (SCREENWIDTH << hires)
+  if (first_allocation) {
+    first_allocation = false;
+
+          solidcol = Z_Malloc(width       * sizeof(byte),    PU_VIDEO, NULL);
+      xtoviewangle = Z_Malloc((width + 1) * sizeof(angle_t), PU_VIDEO, NULL);
+    linearskyangle = Z_Malloc((width + 1) * sizeof(angle_t), PU_VIDEO, NULL);
+  }
+  else {
+          solidcol = Z_Realloc(solidcol,       width       * sizeof(byte),    PU_VIDEO, NULL);
+      xtoviewangle = Z_Realloc(xtoviewangle,   (width + 1) * sizeof(angle_t), PU_VIDEO, NULL);
+    linearskyangle = Z_Realloc(linearskyangle, (width + 1) * sizeof(angle_t), PU_VIDEO, NULL);
+  }
+  #undef width
+  
   R_InitData();
   R_SetViewSize(screenblocks);
   R_InitPlanes();
   R_InitLightTables();
   R_InitSkyMap();
   R_InitTranslationTables();
+  R_InitSprites(sprnames); // [Nugget] Moved from P_Init()
```

Comment:
I moved it because of `*negonearray`. I _could_ have just initialized it (and every other array) in `R_Init()`, but I think it's cleaner if they're initialized in the init function of the module where they're declared.

Reply:
But `negonearray` is really just an array full of `-1`. I don't see much value in having this one allocated dynamically. I would just leave it static and save the hacks.

Reply:
Right, makes sense.